internal/jam/jam.go

```
@@ -34,41 +40,83 @@ type Jam struct {
 	Name     string    `json:"name,omitempty"`
 	Capacity uint      `json:"capacity,omitempty"`
 	BPM      uint      `json:"bpm,omitempty"`
+
+	cli *websocket.Client
 }
 
-func (j *Jam) UnmarshalJSON(data []byte) error {
-	type Alias Jam
-	aux := &struct {
-		*Alias
-	}{
-		Alias: (*Alias)(j),
+// NOTE this should not be empty but panic if it is
+func (j *Jam) Client() *websocket.Client {
+	if j.cli == nil {
+		j.cli = websocket.NewClient(j.Capacity)
 	}
 
-	if err := json.Unmarshal(data, &aux); err != nil {
-		return err
-	}
+	return j.cli
+}
 
-	if aux.Capacity == 0 {
-		j.Capacity = 10
-	}
+func (j *Jam) Close() error {
+	return j.cli.Close()
+}
 
-	if aux.BPM == 0 {
-		j.BPM = 120
+// NOTE -- should init on creation as this is just spinning up excessive goroutines
+func (j *Jam) SetClient(cli *websocket.Client) {
+	if j.cli == nil {
```

Do you want to return an error if the Jam already has a client?
It may be confusing behavior if you call `j.SetClient(..)` but it actually doesn't set the given client. The caller might not even know nothing happen until much later

---

I think this may be safe to remove, was from an older design but not referenced anywhere, yes I think your reasoning is correct